### PR TITLE
Added Custom UserDetails and Utility for  #1730

### PR DESCRIPTION
**Pull request by amit2103**
Added Custom UserDetails and Utility for #1730 
Created the initial code. IS there any development/contribution guide line for JHipster that I can go through? It will be great if someone reviews this code.

### CODE REVIEW

**Comment by gmarziou** [comment]
Yes we have these guidelines: https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md

@deepu105 could you have a look as you were involved in #2305 ?

**Comment by gzsombor** [comment]
The id of the user is not string, for the SQL databases :(

File: app/templates/src/main/java/package/security/_SecurityUtils.java [hunk 1]
```diff
@@ -36,6 +36,22 @@ public static String getCurrentLogin() {
     }
 
     /**
+     * Get the UserDetails associated with the Current User.
+     */
+    public static UserDetails getCurrentLoginId() {
```
Comment (jdubois) [inline app/templates/src/main/java/package/security/_SecurityUtils.java]: The name of the method isn't good, it doesn't get the login id, but a user details

File: app/templates/src/main/java/package/security/_UserPrincipal.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+
+import java.util.List;
+
+/**
+ * Custom implementation of Spring UserDetails
```
Comment (jdubois) [inline app/templates/src/main/java/package/security/_UserPrincipal.java]: Need a dot at the end for correct JavaDoc

File: app/templates/src/main/java/package/security/_UserPrincipal.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    private String password;
+    private List<GrantedAuthority> grantedAuthorities;
+    private String userId;
+
+    /**
+     * Instantiate a Custom UserPrincipal
+     */
+     public UserPrincipal(String login, String password, List<GrantedAuthority> grantedAuthorities , String userId) {
+         this.login = login;
+         this.password = password;
```
Comment (jdubois) [inline app/templates/src/main/java/package/security/_UserPrincipal.java]: Do we really have to store the password here?

File: app/templates/src/main/java/package/security/_UserPrincipal.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+         this.grantedAuthorities = grantedAuthorities;
+         this.userId = userId;
+     }
+
+     public String getLogin() {
+        return login;
+     }
+
+     public String getPassword() {
```
Comment (jdubois) [inline app/templates/src/main/java/package/security/_UserPrincipal.java]: We should not get the password

File: app/templates/src/main/java/package/security/_UserPrincipal.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+     }
+
+     public List<GrantedAuthority> getGrantedAuthorities() {
+          return grantedAuthorities;
+     }
+
+     public String getUserId() {
+         return userId;
+     }
+
```
Comment (jdubois) [inline app/templates/src/main/java/package/security/_UserPrincipal.java]: 2 empty lines should be removed

**Comment by jdubois** [comment]
I added also some comments on @gzsombor 's PR -> I would rather use @gzsombor 's PR, but the difference between both isn't big

**Comment by amit2103** [comment]
@gzsombor  many thanks for the pointer. I had missed that. I checked User class again. Made changes in index.js ( I took the idea from your commit).
@jdubois Thanks for the feedback. I have made the changes. Do a review when you have time. 
If you think @gzsombor  PR is more useful feel free to discard this :)

**Comment by jdubois** [comment]
Yes I'm using the PR from @gzsombor , but mainly because he did it first, so it's more fair to him.

But that would otherwise have been merged, of course.

**Pull request closed without merging**